File: 프로그래머스/3/43162. 네트워크/네트워크.py
def dfs(computers,v,node):
    v[node] = True
    for idx,j in enumerate(computers[node]):
        if j==1 and not v[idx]:
            dfs(computers,v,idx)

def solution(n, computers):
    answer = 0
    v= [False] *n
    
    for i in range(len(v)):
        if not v[i]:
            dfs(computers,v,i)
            answer += 1
    
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    var answer = 0;
    
    // visited arr 정의
    let v = Array(n).fill(false);
    
    // 노드 개수 만큼 반복
    for(let i=0; i<n; i++){
        if(!v[i]){
            answer++;
            dfs(i);
        }
    }
    
    function dfs(index){
        // 방문 완료
        v[index] = true;
        
        const computer = computers[index];
        
        for(let i=0; i<computer.length; i++){
            const isConnect = computer[i] === 1 ? true : false;
            if(!v[i] && isConnect){
                dfs(i)
            }
        }
        
    }
    
    
    return answer;
}